tests/end-to-end/director/labeling_api_test.go

```diff
@@ -0,0 +1,508 @@
+		require.NoError(t, err)
+
+		t.Log("Set label on application")
+		validLabelValue := map[string]interface{}{"foo": "test"}
+
+		setLabelRequest := fixSetApplicationLabelRequest(app.ID, labelKey, validLabelValue)
+		label := graphql.Label{}
+
+		err = tc.RunQuery(ctx, setLabelRequest, &label)
+		require.NoError(t, err)
+		defer deleteLabelDefinition(t, ctx, labelKey, false)
+		defer deleteApplicationLabel(t, ctx, app.ID, labelKey)
+
+		t.Log("Try to delete Label Definition while it's being used by some labels")
+
+		deleteLabelDefinitionRequest := fixDeleteLabelDefinition(labelKey, false)
+		err = tc.RunQuery(context.Background(), deleteLabelDefinitionRequest, nil)
+		require.Error(t, err)
+		assert.EqualError(t, err, "graphql: could not delete label definition, it is already used by at least one label")
+	})
+
+	t.Run("Delete Label from application, than delete the Label Definition - should succeed", func(t *testing.T) {
+
+		t.Log("Create application")
+		app := createRandomApplication(t, ctx, "app")
+		defer deleteApplication(t, app.ID)
+
+		t.Log("Create LabelDefinition")
+		createLabelDefinitionRequest := fixCreateLabelDefinitionRequest(ldInputGql)
+		ld := graphql.LabelDefinition{}
+
+		err = tc.RunQuery(ctx, createLabelDefinitionRequest, ld)
+		require.NoError(t, err)
+
+		t.Log("Set label on application")
+		validLabelValue := map[string]interface{}{labelKey: "test"}
+
+		setLabelRequest := fixSetApplicationLabelRequest(app.ID, labelKey, validLabelValue)
+		label := graphql.Label{}
+
+		err = tc.RunQuery(ctx, setLabelRequest, &label)
+		require.NoError(t, err)
+
+		deleteApplicationLabelRequest := fixDeleteApplicationLabel(app.ID, labelKey)
+		label = graphql.Label{}
+
+		err := tc.RunQuery(ctx, deleteApplicationLabelRequest, &label)
+		require.NoError(t, err)
+		assert.Equal(t, labelKey, label.Key)
+
+		deleteLabelDefinitionRequest := fixDeleteLabelDefinition(labelKey, false)
+		labelDefinition := graphql.LabelDefinition{}
+
+		err = tc.RunQuery(context.Background(), deleteLabelDefinitionRequest, &labelDefinition)
+		require.NoError(t, err)
+		assert.ObjectsAreEqualValues(labelDefinitionInput.Schema, labelDefinition.Schema)
+
+	})
+
+	//TODO write case for deletion of Label Definition with deleting related labels after https://github.com/kyma-incubator/compass/issues/126 is merged
+}
+
+func TestDeleteScenarioLabel(t *testing.T) {
+	// GIVEN
+	ctx := context.Background()
+	t.Log("Create application")
+	app := createRandomApplication(t, ctx, "app")
+	defer deleteApplication(t, app.ID)
+
+	t.Log("Try to delete scenario label on application")
+	labelKey := "scenarios"
+	deleteApplicationLabelRequest := fixDeleteApplicationLabel(app.ID, labelKey)
+
+	// WHEN
+	err := tc.RunQuery(ctx, deleteApplicationLabelRequest, nil)
+
+	//THEN
+	require.Error(t, err)
+	assert.EqualError(t, err, "graphql: scenarios label can not be deleted from application")
+}
+
+func TestDeleteDefaultValueInScenarioLabelDefinition(t *testing.T) {
```

Comment:
Hmm, what about test for modyfying `scenarios` LabelDefinition?

Reply:
added test covering this case: `TestUpdateScenariosLabelDefinitionValue`